server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java

```diff
@@ -168,10 +171,33 @@ public static ServerAddress startServer(MetricsSystem metricsSystem, ServerConte
           addresses);
     } catch (TTransportException e) {
       if (portSearch) {
+        // Build a list of reserved ports - as identified by properties of type PropertyType.PORT
+        ArrayList<Integer> restrictedPorts = new ArrayList<Integer>();
+        for (Property p : Property.values()) {
+          if (p.getType().equals(PropertyType.PORT)) {
+            for (int element : config.getPort(p)) {
+              restrictedPorts.add(element);
+              if (log.isDebugEnabled()) {
+                log.debug("Adding port {} to list of restricted ports", element);
+              }
+            }
+          }
+        }
+
         HostAndPort last = addresses[addresses.length - 1];
         // Attempt to allocate a port outside of the specified port property
         // Search sequentially over the next 1000 ports
         for (int port = last.getPort() + 1; port < last.getPort() + 1001; port++) {
+          if (restrictedPorts.contains(port)) {
+            if (log.isDebugEnabled()) {
+              log.debug(
+                  "During port search, skipping port {} as it is reserved for another service",
+                  port);
+            }
```

Comment:
There is no need to guard log messages with `.isLEVELEnabled()` calls when merely printing a log statement, since the logging framework already does this itself (it's also why we use the format string instead of using `String.format` or string concatenation ourselves). We only need to do that if we're doing extra work to prepare the log messages.

See previous comment on suggestion to improve this log message with the specific service that has reserved it. Could print `Property.getKey()` here.

Reply:
Thanks, I was doing this based on suggestions to do so from @keith-turner for a previous PR - but I realize now that the suggestion was related to the extra work in that case (joining a list) to prep the log message. Will rework.